### DISCUSSION TITLE
Squeeze and Excitation blocks for image classification

### DISCUSSION THREAD

**Discussion by Prajwal111299** [community/community#48275] [Programming Help]
does anyone have some experience in squeeze and excitation networks (SE Blocks) in image classification? I'm getting an uncommon error, pls help.
I'm taking datasets of sharp, blurred, and overexposed images, and classifying them into their subcategories. I'm using Tensorflow but as I progress further I got this error at epoch 1.
![Screenshot (78)](https://user-images.githubusercontent.com/90971007/220636280-0f961316-017c-47a3-bbed-7e38a1800f82.png)
![Screenshot (79)](https://user-images.githubusercontent.com/90971007/220636815-034b3313-0904-4bef-9de9-425e04e8488c.png)

I tried to troubleshoot this but, it seems like there's a problem with my dataset input.
Please help!

**Comment by ademakdogan** [comment]
I faced the same error and tried to test everything with no value, but I heard that you have to make the number of folders in the dataset the SAME as the one in Dense.

I don't know if this will solve your specific bug or not but try this with your code:

model.add(tf.keras.layers.Dense(10, activation = 'softmax'))

Replace 10 with the number of training dataset folders or can call 'classes'.

**Comment by ali-workshop** [comment]
The error you're encountering with Squeeze and Excitation (SE) Blocks in your image classification project using TensorFlow suggests there might be an issue with how the SE Blocks are implemented or integrated into your model. SE Blocks are a mechanism that allows the network to perform feature recalibration, enabling it to learn to selectively emphasize informative features and suppress less useful ones. This is particularly useful in tasks like image classification, where the model needs to focus on relevant features for accurate classification.

Given the error message and the context of your project, here are some steps and considerations to troubleshoot and potentially resolve the issue:

Verify SE Block Implementation: Ensure that the SE Block is correctly implemented according to the TensorFlow documentation or the example provided in the blog post. The SE Block involves two main steps: squeezing the input tensor to generate channel-wise statistics and then exciting the input tensor based on these statistics. The implementation should match the structure and logic outlined in the sources [3](https://www.tensorflow.org/api_docs/python/tfm/vision/layers/SqueezeExcitation)[4](https://blog.paperspace.com/channel-attention-squeeze-and-excitation-networks/).
Check Input and Output Dimensions: The error might be related to mismatched dimensions between the input and output tensors of the SE Block. Ensure that the number of input filters (in_filters) matches the expected input tensor shape and that the number of output filters (out_filters) is correctly specified. The se_ratio parameter, which controls the reduction ratio for the squeeze operation, should also be appropriately set [3](https://www.tensorflow.org/api_docs/python/tfm/vision/layers/SqueezeExcitation).
Review Dataset Input: Since you mentioned the error seems to be related to your dataset input, double-check that your dataset is correctly preprocessed and formatted for input into the model. This includes ensuring that the images are correctly resized, normalized, and batched as expected by the model.
Update TensorFlow Version: Ensure that you are using a compatible version of TensorFlow that supports the features and functions you are using, including the SE Block. TensorFlow's API and functionalities can evolve, and using an outdated version might lead to compatibility issues.
Consult TensorFlow Documentation and Community: The TensorFlow documentation provides detailed information on implementing SE Blocks using the tfm.vision.layers.SqueezeExcitation layer. Additionally, searching for or asking questions on TensorFlow forums or GitHub discussions can provide insights from others who have faced similar issues [3](https://www.tensorflow.org/api_docs/python/tfm/vision/layers/SqueezeExcitation).
Debugging and Logging: Add logging statements before the point of failure to print out tensor shapes and other relevant information. This can help identify where the dimensions or data types might not be as expected.
Simplify the Model: Temporarily simplify your model by removing the SE Blocks and other complex layers to isolate the issue. If the error disappears, gradually reintroduce components to pinpoint the exact cause.
see the link below 
https://stackoverflow.com/questions/64113312/error-when-training-tensorflow-image-classification-model-even-when-all-images-i
